### PR TITLE
Finish wiring up scan().

### LINKED ISSUE #44: Scan
Implement tx.scan -- see https://github.com/rocicorp/replicache-client/blob/master/db/scan.go

### PR DESCRIPTION
**Pull request by aboodman**
Fixes #44.

### CODE REVIEW

File: src/embed/types.rs [hunk 1]
```diff
@@ -90,6 +92,87 @@ pub struct GetResponse {
     pub has: bool, // Second to avoid trailing comma if value == None.
 }
 
+#[derive(DeJson)]
+pub struct ScanKey {
+    value: String,
```
Comment (aboodman) [inline src/embed/types.rs:97]: We use `String` for keys and values at the bindings layer but `Vec<u8>` at key/value layer. We should figure this out someday.

File: src/embed/types.rs [hunk 2]
```diff
@@ -90,6 +92,87 @@ pub struct GetResponse {
...
+    exclusive: bool,
+}
+
+impl<'a> From<&'a ScanKey> for db::ScanKey<'a> {
+    fn from(source: &'a ScanKey) -> db::ScanKey<'a> {
+        db::ScanKey {
+            value: source.value.as_bytes(),
```
Comment (aboodman) [inline src/embed/types.rs:104]: Note: doesn't allocate! whee!

File: src/embed/types.rs [hunk 3]
```diff
@@ -90,6 +92,87 @@ pub struct GetResponse {
...
+#[derive(Debug)]
+pub enum FromProllyEntryError {
+    BadUTF8(std::string::FromUtf8Error),
+}
+
+impl<'a> std::convert::TryFrom<prolly::Entry<'a>> for ScanItem {
+    fn try_from(entry: prolly::Entry) -> Result<Self, Self::Error> {
+        use FromProllyEntryError::*;
+        Ok(Self {
+            key: String::from_utf8(entry.key.to_vec()).map_err(BadUTF8)?,
```
Comment (aboodman) [inline src/embed/types.rs:164]: Note: does allocate :(. Boo.

File: src/embed/types.rs [hunk 4]
```diff
@@ -90,6 +92,87 @@ pub struct GetResponse {
     pub has: bool, // Second to avoid trailing comma if value == None.
 }
 
+#[derive(DeJson)]
```
Comment (phritz) [inline src/embed/types.rs:95]: I think i would have expected to see some dispatch tests for this in https://github.com/rocicorp/repc/blob/master/tests/wasm.rs. Unless there is a technical reason why we can't have them seems like in addition to the scan unit tests in db we probably want some coverage at the higher level as well.

File: src/embed/types.rs (hunk 1 above)
Comment (phritz) [reply src/embed/types.rs:97]: heh

**Review by phritz — COMMENTED** [review]
LGTM

File: src/embed/types.rs (hunk 4 above)
Comment (aboodman) [reply src/embed/types.rs:95]: You got it.

**Pull request merged**